pkg/operator/ceph/csi/peermap/clientProfile.go

```diff
@@ -0,0 +1,160 @@
+	existingClientProfileMappings := &csiopv1a1.ClientProfileMapping{}
+	request := types.NamespacedName{Name: clientProfileMappings.Name, Namespace: clusterInfo.Namespace}
+	err := clusterContext.Client.Get(clusterInfo.Context, request, existingClientProfileMappings)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			err := clusterContext.Client.Create(clusterInfo.Context, clientProfileMappings)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create clientProfileMappings CR %q", clientProfileMappings.Name)
+			}
+			logger.Infof("successfully created clientProfileMappings CR %q", clientProfileMappings.Name)
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get existing clientProfileMappings CR %q", clientProfileMappings.Name)
+	}
+	existingClientProfileMappings.Spec = clientProfileMappings.Spec
+	err = clusterContext.Client.Update(clusterInfo.Context, existingClientProfileMappings)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update existing clientProfileMappings CR  %q", clientProfileMappings.Name)
+	}
+	logger.Infof("successfully updated clientProfileMappings CR %q", clientProfileMappings.Name)
+
+	return nil
+}
+
+func generateClientProfileMappingsCR(clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, pool *cephv1.CephBlockPool) (*csiopv1a1.ClientProfileMapping, error) {
+	mappings, err := getClusterPoolIDMap(clusterContext, clusterInfo, pool)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get peer pool ID mappings for the pool %q", pool.Name)
+	}
+
+	blockPoolMappings := []csiopv1a1.BlockPoolMappingSpec{}
+	for _, mapping := range *mappings {
+		remoteClusterID, localClusterID := getFirstKeyValue(mapping.ClusterIDMapping)
+		for _, poolIDMappings := range mapping.RBDPoolIDMapping {
+			remotePoolID, localPoolID, err := getFirstKeyValueAsInt(poolIDMappings)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get pool IDs as integers")
+			}
+			blockPoolMapping := csiopv1a1.BlockPoolMappingSpec{
+				Local:  csiopv1a1.BlockPoolRefSpec{ClientProfileName: localClusterID, PoolId: localPoolID},
+				Remote: csiopv1a1.BlockPoolRefSpec{ClientProfileName: remoteClusterID, PoolId: remotePoolID},
+			}
+			logger.Debugf("cluster ID mappings: %q:%q", remoteClusterID, localClusterID)
+			logger.Debugf("pool ID mappings: %d:%d", remotePoolID, localPoolID)
+			blockPoolMappings = append(blockPoolMappings, blockPoolMapping)
+		}
+	}
+
+	clientProfileMappings := &csiopv1a1.ClientProfileMapping{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pool.Name, //TODO: what should be the correct name for this resource?
+			Namespace: clusterInfo.Namespace,
+		},
+		Spec: csiopv1a1.ClientProfileMappingSpec{
+			BlockPoolMapping: blockPoolMappings,
+		},
+	}
+
+	err = clusterInfo.OwnerInfo.SetOwnerReference(clientProfileMappings)
```

Comment:
Could we consider the pool as the owner of this mapping? If so, then the mapping CR will be automatically deleted when the pool is deleted.

Reply:
Yes. That would work. Added CephBlockPool as the owner of the resource and removed the explicit call to delete the clientProfilemappings resource when cephBlockPool is deleted.